pydantic/types.py

```diff
@@ -1077,6 +1083,172 @@ class ConstrainedExample(BaseModel):
+print(repr(m))
+#> Model(negative_decimal=Decimal('-1.0'))
+
+try:
+    Model(negative_decimal=Decimal('1.0'))
+except ValidationError as e:
+    print(e.errors())
+    '''
+    [
+        {
+            'type': 'less_than',
+            'loc': ('negative_decimal',),
+            'msg': 'Input should be less than 0',
+            'input': Decimal('1.0'),
+            'ctx': {'lt': Decimal('0')},
+            'url': 'https://errors.pydantic.dev/2/v/less_than',
+        }
+    ]
+    '''
+```
+"""
+NonPositiveDecimal = Annotated[Decimal, annotated_types.Le(0)]
+"""
+A decimal that must be less than or equal to zero.
+
+```py
+from decimal import Decimal
+from pydantic import BaseModel, NonPositiveDecimal, ValidationError
+
+class Model(BaseModel):
+    non_positive_decimal: NonPositiveDecimal
+
+m = Model(non_positive_decimal=Decimal('0.0'))
+print(repr(m))
+#> Model(non_positive_decimal=Decimal('0.0'))
+
+try:
+    Model(non_positive_decimal=Decimal('1.0'))
+except ValidationError as e:
+    print(e.errors())
+    '''
+    [
+        {
+            'type': 'less_than_equal',
+            'loc': ('non_positive_decimal',),
+            'msg': 'Input should be less than or equal to 0',
+            'input': Decimal('1.0'),
+            'ctx': {'le': Decimal('0')},
+            'url': 'https://errors.pydantic.dev/2/v/less_than_equal',
+        }
+    ]
+    '''
+```
+"""
+NonNegativeDecimal = Annotated[Decimal, annotated_types.Ge(0)]
+"""A decimal that must be greater than or equal to zero.
+
+```py
+from decimal import Decimal
+from pydantic import BaseModel, NonNegativeDecimal, ValidationError
+
+class Model(BaseModel):
+    non_negative_decimal: NonNegativeDecimal
+
+m = Model(non_negative_decimal=Decimal('0.0'))
+print(repr(m))
+#> Model(non_negative_decimal=Decimal('0.0'))
+
+try:
+    Model(non_negative_decimal=Decimal('-1.0'))
+except ValidationError as e:
+    print(e.errors())
+    '''
+    [
+        {
+            'type': 'greater_than_equal',
+            'loc': ('non_negative_decimal',),
+            'msg': 'Input should be greater than or equal to 0',
+            'input': Decimal('-1.0'),
+            'ctx': {'ge': Decimal('0')},
+            'url': 'https://errors.pydantic.dev/2/v/greater_than_equal',
+        }
+    ]
+    '''
+```
+"""
+StrictDecimal = Annotated[Decimal, Strict()]
+"""A decimal that must be validated in strict mode.
+
+```py
+from pydantic import BaseModel, StrictDecimal, ValidationError
+
+class StrictDecimalModel(BaseModel):
+    strict_decimal: StrictDecimal
+
+try:
+    StrictDecimalModel(strict_decimal='1.0')
+except ValidationError as e:
+    print(e)
+    '''
+    1 validation error for StrictDecimalModel
+    strict_decimal
+      Input should be an instance of Decimal [type=is_instance_of, input_value='1.0', input_type=str]
+    '''
+```
+"""
+InfiniteDecimal = Annotated[Decimal, AllowInfNan(True)]
+"""A decimal that can be infinite or NaN(``-Infinity``, ``Infinity``, or ``nan``).
```

Comment:
```suggestion
"""A decimal that can be infinite or NaN(`-Infinity`, `Infinity`, or `nan`).
```
> Regarding the docs, can you elaborate on the md/rst fix needed? thank you

I think it was regarding double quotes

Reply:
Done